src/hera/workflows/workflow.py

```diff
@@ -3,15 +3,20 @@
 See https://argoproj.github.io/argo-workflows/workflow-concepts/#the-workflow
 for more on Workflows.
 """
-from __future__ import annotations
+from inspect import get_annotations
```

Comment:
`get_annotations` was introduced in python 3.10 so we'd have to copy the function and dependencies over, as we did for `unparse`.